presto-main/src/main/java/com/facebook/presto/cost/ScalarStatsCalculator.java

```diff
@@ -91,5 +96,62 @@ protected SymbolStatsEstimate visitLiteral(Literal node, Void context)
                     .setDistinctValuesCount(1)
                     .build();
         }
+
+        @Override
+        protected SymbolStatsEstimate visitArithmeticBinary(ArithmeticBinaryExpression node, Void context)
+        {
+            requireNonNull(node, "node is null");
+            checkArgument(node.getType() != ArithmeticBinaryExpression.Type.MODULUS, "Modulus operator not yet supported");
+            SymbolStatsEstimate left = process(node.getLeft());
+            SymbolStatsEstimate right = process(node.getRight());
+
+            SymbolStatsEstimate.Builder result = SymbolStatsEstimate.builder()
+                    .setDataSize(Math.max(left.getDataSize(), right.getDataSize()))
+                    .setNullsFraction(left.getNullsFraction() + right.getNullsFraction())
+                    .setDistinctValuesCount(left.getDistinctValuesCount() * right.getDistinctValuesCount());
+
+            double leftLow = left.getLowValue();
+            double leftHigh = left.getHighValue();
+            double rightLow = right.getLowValue();
+            double rightHigh = right.getHighValue();
+            if (isInfinity(leftLow) || isInfinity(leftHigh) || isInfinity(rightLow) || isInfinity(rightHigh)) {
+                result.setLowValue(Double.MIN_VALUE)
+                        .setHighValue(Double.MAX_VALUE);
+            }
+            else {
+                double v1 = operate(node.getType(), leftLow, rightLow);
+                double v2 = operate(node.getType(), leftLow, rightHigh);
+                double v3 = operate(node.getType(), leftHigh, rightLow);
+                double v4 = operate(node.getType(), leftHigh, rightHigh);
+
+                result.setLowValue(min(v1, min(v2, min(v3, v4))))
+                        .setHighValue(max(v1, max(v2, max(v3, v4))));
+            }
+
+            return result.build();
+        }
+
+        private boolean isInfinity(double value)
```

Comment:
drop this.
`double` actually can store infinity values. There is `Double.POSITIVE_INFINITY` and `Double.NEGATIVE_INFINITY` and static methods in `Double` to check that (`isFinite`, `isInfinite` ...)